Tasks/AzureRmWebAppDeployment/operations/KuduServiceUtility.ts

```diff
@@ -127,6 +127,56 @@ export class KuduServiceUtility {
         }
     }
 
+    public async zipDeploy(packagePath: string, appOffline?: boolean, customMessage?: any): Promise<void> {
+        try {
+            
+            if(tl.stats(packagePath).isDirectory()) {
+                let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                packagePath = await zipUtility.archiveFolder(packagePath, "", tempPackagePath);
+                tl.debug("Compressed folder " + packagePath + " into zip : " +  packagePath);
+            }
+
+            if(appOffline) {
+                await this._appOfflineKuduService(physicalRootPath, true);
+                tl.debug('Wait for 10 seconds for app_offline to take effect');
+                await webClient.sleepFor(10);
+            }
+
+            var deploymentRequestDetails = this._getUpdateHistoryRequest(true, null, customMessage);
+            var queryParameters: Array<string> = [
+                'isAsync=true',
+                'message=' + encodeURIComponent(deploymentRequestDetails.message),
+                'deployer=' + 'VSTS',
```

Comment:
Do we want to set the deployer to VSTS here ? I am assuming this will be used for the log pushed as a part of zip deploy.  It will be better to retain deployer as default 'ZipDeploy' so that we can filter it out in Ibiza CD flow.

Reply:
changed to `VSTS_ZIP_DEPLOY`. thanks